R/data-set.R

```diff
@@ -51,76 +60,166 @@ DataSet <- R6::R6Class(
       }
       private$.yValues$displayUnit <- value
     },
-    #' @field yValues Values stored in the yUnit dimension (not necessarily in the base unit of the dimension)
+    #' @field yValues Values stored in the yDimension dimension (not necessarily in the base unit)
     yValues = function(values) {
       if (missing(values)) {
-        return(private$getColumnValues(private$.yValues))
+        return(private$.getColumnValues(private$.yValues))
+      }
+      validateIsSameLength(self$xValues, values)
+      private$.setColumnValues(private$.yValues, values)
+    },
+
+    #' @field yErrorType Type of the error - geometric or arithmetic.
+    #' When changing from arithmetic to geometric error, the values are considered in as fraction (1 = 100%).
+    #' When changing from geometric to arithmetic, the values are set to the same unit as \code{yErrorUnit}.
+    yErrorType = function(value) {
+      if (missing(value)) {
+        dataInfo <- rClr::clrGet(private$.yErrorValues$ref, "DataInfo")
+        errorTypeEnumVal <- rClr::clrGet(dataInfo, "AuxiliaryType")
+        return(netEnumName("OSPSuite.Core.Domain.Data.AuxiliaryType", errorTypeEnumVal))
+      }
+      private$.setErrorType(value)
+    },
+    #' @field yErrorUnit Unit in which the yErrorValues are defined. For arithmetic error, the unit must be valid
+    #' for \code{yDimension}. For geometric error, the unit must be valid for \code{Dimensionless}.
+    yErrorUnit = function(value) {
+      if (missing(value)) {
+        return(private$.yErrorValues$displayUnit)
+      }
+      private$.yErrorValues$displayUnit <- value
+    },
+    #' @field yErrorValues Values of error stored in the yErrorUnit unit
+    yErrorValues = function(values) {
+      if (missing(values)) {
+        return(private$.getColumnValues(private$.yErrorValues))
       }
 
-      private$setColumnValues(private$.yValues, values)
+      validateIsSameLength(self$yValues, values)
+      private$.setColumnValues(private$.yErrorValues, values)
+    },
+
+    #' @field metaData Returns a named list of meta data defined for the data set.
+    metaData = function(value) {

```

Comment:
We should be able to get meta data but also add entries. This is the proposed solution where `metaData` just returns a list of all entries.